### PR TITLE
CLI: Add --no-open flag

### PR DESCRIPTION
**Pull request by yannbf**
Issue: #6201

## What I did

This PR adds a flag to enable not opening the browser.
## How to test

- `git fetch && git checkout feature/cli-no-open-flag`
- `yarn build core-server`

From there, try it out
**Should open Storybook on the browser:**
- yarn start

**Should not open Storybook on the browser:**
- yarn start --no-open
- yarn start --ci
- yarn start --smoke-test

--- 
- Is this testable with Jest or Chromatic screenshots? no
- Does this need a new example in the kitchen sink apps? no
- Does this need an update to the documentation? yess

### CODE REVIEW

**Review by shilman — APPROVED** [review]
Awesome. Thanks @yannbf ❤️

File: lib/core-server/src/cli/dev.ts [hunk 1]
```diff
@@ -29,6 +29,7 @@ export async function getDevCli(packageJson: {
     .option('--ssl-key <key>', 'Provide an SSL key. (Required with --https)')
     .option('--smoke-test', 'Exit after successful start')
     .option('--ci', "CI mode (skip interactive prompts, don't open browser)")
+    .option('--no-open', 'Do not open Storybook automatically in the browser')
```
Comment (shilman) [inline lib/core-server/src/cli/dev.ts:32]: Should this be `--open`? @yannbf

**Review by shilman — CHANGES_REQUESTED** [review]

File: lib/core-server/src/cli/dev.ts (hunk 1 above)
Comment (yannbf) [reply lib/core-server/src/cli/dev.ts:32]: According to [commander's docs](https://github.com/tj/commander.js/#other-option-types-negatable-boolean-and-booleanvalue) that's how you set a flag that is true by default and will be set to false when used. With this option set, `--open` will not be recognized, only `--no-open`. However, when would anyone ever use the flag `--open` if that is true by default? 🤔

Comment (shilman) [reply lib/core-server/src/cli/dev.ts:32]: 🚀

**Review by shilman — COMMENTED** [review]
LGTM!

**Review by shilman — APPROVED** [review]

**Pull request merged**